File: server/models/MonitoringSite.js
const mongoose = require('mongoose');

const monitoringSiteSchema = new mongoose.Schema({
  siteName: {
    type: String,
    required: true
  },
  siteCode: {
    type: String,
    required: true,
    unique: true
  },
  habitatType: {
    type: String,
    enum: ['Forest', 'Grassland', 'Wetland', 'Desert', 'Mountain', 'Other'],
    default: 'Forest'
  },
  protectedArea: {
    type: String 
  },
  // Optional. Needed to compute real population DENSITY (individuals / km²).
  // If left unset, density is reported as unavailable for that site rather
  // than estimated or faked.
  areaKm2: {
    type: Number
  },
  location: {
    latitude: { type: Number, required: true },
    longitude: { type: Number, required: true }
  },
  monitoringDevice: {
    type: String,
    enum: ['Camera Trap', 'Manual Observation', 'Other'],
    default: 'Camera Trap'
  },
  registeredBy: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  active: {
    type: Boolean,
    default: true
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('MonitoringSite', monitoringSiteSchema);

File: backend/app/schemas/monitoring_site.py
from pydantic import BaseModel
from typing import Optional
from pydantic import BaseModel


class MonitoringSiteCreate(BaseModel):
    survey_id: str
    site_name: str
    latitude: float
    longitude: float
    habitat_type: str
    protected_area: str | None = None
    monitoring_device: str | None = None


class MonitoringSiteResponse(BaseModel):
    id: str
    survey_id: str
    site_name: str
    latitude: float
    longitude: float
    habitat_type: str
    protected_area: str | None
    monitoring_device: str | None

    class Config:
        from_attributes = True


class MonitoringSiteUpdate(BaseModel):
    site_name: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    habitat_type: Optional[str] = None
    protected_area: Optional[str] = None
    monitoring_device: Optional[str] = None

